dso-l2/src/main/java/com/tc/objectserver/handler/ProcessTransactionHandler.java

```diff
@@ -106,6 +108,14 @@ public synchronized void reconnectComplete() {
     public void handleEvent(TCMessage context) throws EventHandlerException {
       NodeID destinationID = context.getDestinationNodeID();
       invokeReturn.remove((ClientID)destinationID, context);
+      if(context instanceof VoltronEntityMultiResponse) {
+        VoltronEntityMultiResponse voltronEntityMultiResponse = (com.tc.entity.VoltronEntityMultiResponse) context;
+        for (TransactionID transactionID : voltronEntityMultiResponse.getReceivedTransactions()) {
+          waitForTransactionOrderPersistenceFuture(transactionID);
+        }
+      } else if(context instanceof VoltronEntityAppliedResponse) {
```

Comment:
This is definitely easier to follow.

Are these the only 2 types of messages which come in here?

If so, either make this a generic else (and document that there are only 2 types, prior to the cast) or add a generic else to document that there are only 2 types and put an assert there.

If not, add a generic else with some comments explaining why you can get there and why not handling the persistence future makes sense, in that case.

As it stands, I am worried that a change to the type will mean we fall out of this, incorrectly.